Eevee.Sleep.Bot/Controllers/Mongo/UserRoleController.cs

```diff
@@ -0,0 +1,58 @@
+
+
+
+using Eevee.Sleep.Bot.Models;
+using MongoDB.Driver;
+
+namespace Eevee.Sleep.Bot.Controllers.Mongo;
+
+public static class UserRoleContoller
+{
+    public static async Task SaveRoles(string userId, ulong[] roles)
+    {
+        var collection = MongoConst.RoleCollection;
+        var filter = Builders<UserRoleModel>.Filter.Eq("userId", userId);
```

Comment:
See if all these filters and updates can be converted to this form
```csharp
var removeActivationKeyTask = MongoConst.AuthActivationKeyCollection.FindOneAndDeleteAsync(
    Builders<ActivationKeyModel>.Filter.Where(x => x.Contact.Discord == userId)
);
```

The biggest difference is that what we currently have is using string constant, if the key name ever changes, this won't get refactored, which is bad. Using something typed allows us to seamlessly refactor it in the future.